src/openrct2-ui/windows/Error.cpp

```diff
@@ -84,96 +131,35 @@ rct_window* WindowErrorOpen(std::string_view title, std::string_view message)
     if (buffer.size() <= 1)
         return nullptr;
 
-    width = gfx_get_string_width_new_lined(buffer.data(), FontStyle::Medium);
+    int32_t width = gfx_get_string_width_new_lined(buffer.data(), FontStyle::Medium);
     width = std::clamp(width, 64, 196);
 
+    int32_t numLines = 1;
     gfx_wrap_string(buffer.data(), width + 1, FontStyle::Medium, &numLines);
 
-    _window_error_num_lines = numLines;
     width = width + 3;
-    height = (numLines + 1) * font_get_line_height(FontStyle::Medium) + 4;
-
-    window_error_widgets[WIDX_BACKGROUND].right = width;
-    window_error_widgets[WIDX_BACKGROUND].bottom = height;
-
+    int32_t height = (numLines + 1) * font_get_line_height(FontStyle::Medium) + 4;
     int32_t screenWidth = context_get_width();
     int32_t screenHeight = context_get_height();
     const CursorState* state = context_get_cursor_state();
     ScreenCoordsXY windowPosition = state->position - ScreenCoordsXY(width / 2, -26);
     windowPosition.x = std::clamp(windowPosition.x, 0, screenWidth);
     windowPosition.y = std::max(22, windowPosition.y);
-    maxY = screenHeight - height;
+    int32_t maxY = screenHeight - height;
     if (windowPosition.y > maxY)
     {
         windowPosition.y = std::min(windowPosition.y - height - 40, maxY);
     }
 
-    w = WindowCreate(
-        windowPosition, width, height, &window_error_events, WindowClass::Error,
-        WF_STICK_TO_FRONT | WF_TRANSPARENT | WF_RESIZABLE);
-    w->widgets = window_error_widgets;
-    w->error.var_480 = 0;
-    if (!gDisableErrorWindowSound)
-    {
-        OpenRCT2::Audio::Play(OpenRCT2::Audio::SoundId::Error, 0, w->windowPos.x + (w->width / 2));
-    }
-
-    return w;
+    auto errorWindow = std::make_unique<ErrorWindow>(buffer, numLines);
+    return static_cast<ErrorWindow*>(WindowCreate(
+        std::move(errorWindow), WindowClass::Error, windowPosition, width, height,
+        WF_STICK_TO_FRONT | WF_TRANSPARENT | WF_RESIZABLE));
```

Comment:
```suggestion
    return WindowCreate(
        std::move(errorWindow), WindowClass::Error, windowPosition, width, height,
        WF_STICK_TO_FRONT | WF_TRANSPARENT | WF_RESIZABLE);
```

I don't think the cast is needed.